lib/kafka_ex/protocol/join_group.ex

```diff
@@ -0,0 +1,43 @@
+defmodule KafkaEx.Protocol.JoinGroup do
+  import KafkaEx.Protocol.Common
+
+  @protocol_type "consumer"
+  @strategy_name "assign"
+  @metadata_version 0
+
+  defmodule Response do
+    defstruct error_code: 0, generation_id: 0, leader: "", members: []
```

Comment:
I'd say use default values of `nil` or `[]`.  `error_code: 0` might be a reasonable default if you're only setting an error if an error occurs.